docs/products/access/api-reference/config-api/identity-provider-api.md

```diff
@@ -93,13 +94,104 @@ Pragma: no-cache
+| user_info_password           | OAUTH        | depends  | Identity source password. <br/> Required when `user_info_enabled` is `true`.                                                                                                                                            |
+
+Example `TULIP` type request:
+
+```http
+POST /api/v1/configuration/idps 
+Host: onegini.example.com
+Content-Type: application/json
+{
+   "id":"tulip-insurcar",
+   "name":"Tulip Insurcar Idp",
+   "type":"TULIP",
+   "client_id":"oauth2CustomerApp",
+   "issuer_uri":"https://insurgroup-edge.iwelcome.com/insurcar/login/oauth2/v1",
+   "end_session_enabled":true,
+   "scopes":"openid",
+   "integrations":[
+      "APP_TO_WEB",
+      "UDH_API"
+   ],
+   "tulip_api_client_id":"accessIntegration",
+   "tulip_api_base_url":"https://insurgroup-edge.iwelcome.com/insurcar",
+   "tulip_api_access_scope":"iwelcome:segment:insurgroup"
+}
+```
+
+Example `OAUTH` type request:
+
+```http
+POST /api/v1/configuration/idps 
+Host: onegini.example.com
+Content-Type: application/json
+{
+  "type": "OAUTH",
+  "id": "oauth-idp",
+  "name": "OAuth Identity Provider",
+  "client_id": "oauth2CustomerApp",
+  "authorization_url": "https://onegini.test.onegini.com/oauth/v1/authorize",
+  "token_url": "https://onegini.test.onegini.com/oauth/v1/token",
+  "profile_url": "https://onegini.test.onegini.com/oauth/v1/userinfo",
+  "scopes": "openid profile email"
+}
+```
+
+Example success response:
+
+```http
+HTTP/1.1 201 CREATED
+Content-Type: application/json;charset=UTF-8
+Cache-Control: no-store
+Pragma: no-cache
+Location: /api/v1/configuration/idps/tulip-insurcar
+```
+
+The success response body is empty.
+
 ## Error codes
 
 One of the following responses will be returned, containing a JSON object with an error code, a message and details about the error.
 
-| HTTP status | Error code                       | Message 
-|-------------|----------------------------------|-------------------------------------------------------------------------------------
-| 400         | invalid_request                  | One or more parameters is missing or incorrect. The details contain the missing or incorrect parameters.
-| 401         | unauthorized                     | Provide valid credentials to get access to the API.
-| 403         | forbidden                        | Operation is not allowed for the current user. 
-| 404         | not_found                        | Identity Provider for given ID cannot be found
+| HTTP status | Error code      | Message                                                                                                  |
+|-------------|-----------------|----------------------------------------------------------------------------------------------------------|
```

Comment:
409 is also technically possible as well if you try to create with the same ID?

Reply:
All validation errors are returned as 400, including duplicated id.